editions/tw5.com/tiddlers/concepts/Customizing EditTemplate Field Rendering.tid

```diff
@@ -0,0 +1,31 @@
+created: 20220305183700000
+modified: 20220305183700000
+tags: Concepts
+title: Customizing EditTemplate field rendering
+type: text/vnd.tiddlywiki
+
+When editing a tiddler the [[EditTemplate|$:/core/ui/EditTemplate/fields]] normally renders fields as simple input boxes. To modify this behaviour, the [[cascade mechanism|Cascades]] can be used. Via the [[Field Editor Cascade|Field Editor Cascade]] the name of the tiddler used for rendering the field editor can be specified. The content of this tiddler is transcluded to represent the content of the field.
+
+To modify the appearance of all fields whose name ends with `-date` create a new tiddler and add the `$:/tags/FieldEditorFilter` tag to it. Add a `list-before` field and assign the value `$:/config/FieldEditorFilters/default`. Now you have to put the filter for the cascade into the tiddler's text field: `[regexp[-date$]then[$:/config/EditTemplateFields/Templates/dates]]`. This will transclude the tiddler named `$:/config/EditTemplateFields/Templates/dates` to render the input elements for all fields with names matching the regular expression.
```

Comment:
Instead of `regexp` we should use the `suffix` filter for the docs. It's easier to understand